### PR TITLE
Feat/added is followed and is liked fields to tweet

### PR DESCRIPTION
**Pull request by DavidKorystov**

### CODE REVIEW

File: src/routes/tweets.route.ts [hunk 1]
```diff
@@ -14,10 +14,12 @@ import {
 
 const tweetsRoutes = express.Router();
 
-tweetsRoutes.route("/").get(validation(getTweets), async (req, res) => {
-  const tweets = await tweetsService.getAllTweets(req);
-  res.send(tweets);
-});
+tweetsRoutes
+  .route("/")
+  .get(isAuthorized, validation(getTweets), async (req, res) => {
```
Comment (VitaliiDenysiuk) [inline src/routes/tweets.route.ts]: Why did you decide close all tweets for not authorized user?  This needs to be agreed with the whole team. If this needed only for "liked" - you can check if req.user?._id ===undefined then you don't do the logic with "liked"

File: src/services/tweets.service.ts [hunk 2]
```diff
@@ -13,18 +14,60 @@ import {
...
 
-export const getAllTweetsOfCurrentUser = (req: Request) => {
+export const getAllTweets = async (req: Request) => {
   const [query, options] = new PaginationParameters({ query: req.query }).get();
-  return getList({ ...query, author: req.user._id }, options);
+  options.leanWithId = true;
+  options.lean = true;
+  const list = await getList(query, options);
+  const listOfTweetId = list.docs.map((doc) => doc.id);
+  console.warn(listOfTweetId);
```
Comment (VitaliiDenysiuk) [inline src/services/tweets.service.ts]:
```suggestion
```
clean that

File: src/services/tweets.service.ts [hunk 3]
```diff
@@ -13,18 +14,60 @@ import {
...
+
+  liked.docs.forEach((record) => {
+    listOfTweetId.forEach((currentId, index) => {
+      if (currentId == record.tweet._id) {
+        list.docs[index].isLiked = true;
+      }
+    });
+  });
+
+  return list;
```
Comment (VitaliiDenysiuk) [inline src/services/tweets.service.ts]:
```suggestion

  return getAllTweets(req, req.user._id);
```
1) you can call function getAllTweets here, 
2) clean all other duplicate code
3) add not required params forUserId for function getAllTweets
4) add logic to getAllTweets to check filled param  forUserId or no

it is very bad to extend two identical functions

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]
very good all logic with isFollowed and isLiked. But take into account some remarks

File: src/routes/tweets.route.ts (hunk 1 above)
Comment (DavidKorystov) [reply src/routes/tweets.route.ts]: Sorry, I didn't know you could view tweets without logging in

**Review by VitaliiDenysiuk — APPROVED** [review]
good

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**